plugin/src/main/java/io/jenkins/plugins/casc/impl/secrets/VaultSecretSource.java

```diff
@@ -39,114 +37,200 @@
-        Optional<String> vaultAppRoleSecret = getVariable(CASC_VAULT_APPROLE_SECRET, prop);
-        Optional<String> vaultNamespace = getVariable(CASC_VAULT_NAMESPACE, prop);
-        Optional<String> vaultEngineVersion = getVariable(CASC_VAULT_ENGINE_VERSION, prop);
-
-        Optional<String[]> vaultPaths = getCommaSeparatedVariables(CASC_VAULT_PATHS, prop)
+        Optional<String> vaultEngineVersionOpt = getVariable(CASC_VAULT_ENGINE_VERSION, prop);
+        Optional<String> vaultUrlOpt = getVariable(CASC_VAULT_URL, prop);
+        Optional<String> vaultNamespaceOpt = getVariable(CASC_VAULT_NAMESPACE, prop);
+        Optional<String> vaultPwOpt = getVariable(CASC_VAULT_PW, prop);
+        Optional<String> vaultUserOpt = getVariable(CASC_VAULT_USER, prop);
+        Optional<String> vaultMountOpt = getVariable(CASC_VAULT_MOUNT, prop);
+        Optional<String> vaultTokenOpt = getVariable(CASC_VAULT_TOKEN, prop);
+        Optional<String> vaultAppRoleOpt = getVariable(CASC_VAULT_APPROLE, prop);
+        Optional<String> vaultAppRoleSecretOpt = getVariable(CASC_VAULT_APPROLE_SECRET, prop);
+        Optional<String[]> vaultPathsOpt = getCommaSeparatedVariables(CASC_VAULT_PATHS, prop)
                 .map(Optional::of)
                 .orElse(getCommaSeparatedVariables(CASC_VAULT_PATH, prop)); // TODO: deprecate!
 
         // Check mandatory variables are set
-        if (!vaultUrl.isPresent() || !vaultPaths.isPresent()) return;
+        if (!vaultUrlOpt.isPresent() || !vaultPathsOpt.isPresent()) return;
 
         // Check defaults
-        if (!vaultMount.isPresent()) vaultMount = Optional.of(DEFAULT_USER_BACKEND);
-        if (!vaultEngineVersion.isPresent()) vaultEngineVersion = Optional.of(DEFAULT_ENGINE_VERSION);
+        vaultMount = vaultMountOpt.orElse(DEFAULT_USER_BACKEND);
+        String vaultEngineVersion = vaultEngineVersionOpt.orElse(DEFAULT_ENGINE_VERSION);
+
+        // Set class fields
+        String vaultUrl = vaultUrlOpt.get();
+        vaultPw = vaultPwOpt.orElse(null);
+        vaultUser = vaultUserOpt.orElse(null);
+        vaultMount = vaultMountOpt.orElse(null);
+        vaultToken = vaultTokenOpt.orElse(null);
+        vaultAppRole = vaultAppRoleOpt.orElse(null);
+        vaultAppRoleSecret = vaultAppRoleSecretOpt.orElse(null);
+        vaultPaths = vaultPathsOpt.orElse(null);
 
         // configure vault client
-        VaultConfig vaultConfig = new VaultConfig().address(vaultUrl.get());
+        vaultConfig = new VaultConfig().address(vaultUrl);
         try {
-            LOGGER.log(Level.FINE, "Attempting to connect to Vault: {0}", vaultUrl.get());
-            if (vaultNamespace.isPresent()) {
-                vaultConfig.nameSpace(vaultNamespace.get());
-                LOGGER.log(Level.FINE, "Using namespace with Vault: {0}", vaultNamespace);
+            LOGGER.log(Level.FINE, "Attempting to connect to Vault: {0}", vaultUrl);
+            if (vaultNamespaceOpt.isPresent()) {
```

Comment:
```suggestion
            vaultNamespaceOpt.ifPresent(namespace ->
```

Reply:
The problem with that, the method can throw an exception so it would be worse 😄